### DISCUSSION TITLE
File Upload: Can I get the URL of an uploaded blob?

### DISCUSSION THREAD

**Discussion by pihai** [Azure/azure-iot-sdk-c#2095] [Ideas]
The .NET SDK gives me the ability to get an absolute URL of the uploaded blob. See the example below:

```csharp
var fileUploadSasUriRequest = new FileUploadSasUriRequest
{
    BlobName = fileName
};
// Lines removed for clarity
FileUploadSasUriResponse sasUri = await _deviceClient.GetFileUploadSasUriAsync(fileUploadSasUriRequest);
Uri uploadUri = sasUri.GetBlobUri();
```

Is there a possibility to get the absolute blob URL via the c SDK? I can't find a way to achieve this with the `IoTHubDeviceClient_LL_UploadToBlob` function. I would like to store the URL on the device or return it via a direct method to the caller.

What I'm also missing from `IoTHubDeviceClient_LL_UploadToBlob` is the ability to add properties/meta-data to the blob.

While the blob upload function is super easy to use it is also somewhat limited. Any recommendation on how I could achieve the requirements described above with the C SDK?

Thanks,
Philipp

**Comment by danewalton-msft** [comment]
Hi @pihai we don't currently have that functionality and it is unlikely that we would add it in the short term. Same with the properties. I will convert this to a discussion as a feature ask to keep in the mind in the future.

As a possible alternative, we do have a [storage SDK for our embedded C SDK](https://github.com/Azure/azure-sdk-for-c/tree/feature/StorageBlobs/sdk/samples/storage/blobs) which would supply greater flexibility to get this to work (albeit you would then be mixing two different SDKs).
